tests/vm_test.go

```diff
@@ -289,15 +289,18 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 			vm := createVM(virtClient, libvmifact.NewCirros())
 
-			err = retryWithMetadataIfModified(vm.ObjectMeta, func(meta k8smetav1.ObjectMeta) error {
-				vm, err = virtClient.VirtualMachine(meta.Namespace).Get(context.Background(), meta.Name, k8smetav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				vm.Spec.Template.ObjectMeta.Annotations = annotations
-				vm, err = virtClient.VirtualMachine(meta.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
-				return err
-			})
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			annotationsPatch, err := patch.New(
+				patch.WithAdd(
+					"/spec/template/spec/metadata/annotations",
+					annotations,
+				)).GeneratePayload()
 			Expect(err).ToNot(HaveOccurred())
 
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, annotationsPatch, metav1.PatchOptions{})
+
+			Expect(err).ToNot(HaveOccurred())
 			vm = startVM(virtClient, vm)
```

Comment:
And now that I see this, it looks very similar to something we already handled in the past.
There is no need for and update or patch, the VM should be created with the annotation in the first place. There is no meaning of updating the VM after the creation and without it running yet... there is no VMI yet.

Please examine if this is the same for the other cases below.